src/core/types/addons.js

```diff
@@ -0,0 +1,67 @@
+/* @flow */
+export type AddonVersionType = {|
+  id: number,
+  channel: string,
+  edit_url: string,
+  files: Array<Object>,
+  // The `text` property is omitted from addon.current_version.license.
+  license: { name: string, url: string },
+  reviewed: Date,
+  version: string,
+|};
+
+export type AddonAuthorType = {|
+  name: string,
+  url: string,
+|};
+
+export type AddonType = {|
+  id: number,
+  authors: Array<AddonAuthorType>,
+  average_daily_users: number,
+  categories: Object,
+  compatibility: Object,
+  current_version: AddonVersionType,
+  default_locale: string,
+  description: string,
+  edit_url: string,
+  guid: string,
+  has_eula: boolean,
+  has_privacy_policy: boolean,
+  homepage: string,
+  icon_url: string,
+  is_disabled: boolean,
+  is_experimental: boolean,
+  is_source_public: boolean,
+  name: string,
+  last_updated: Date,
+  latest_unlisted_version: ?AddonVersionType,
+  previews: Array<Object>,
+  public_stats: boolean,
+  ratings: {|
+    count: number,
+    average: number,
+  |},
+  review_url: string,
+  slug: string,
+  status: 'beta'
+    | 'lite'
```

Comment:
It is a [Union type](https://flow.org/en/docs/types/unions/) so I think it's correct but, huh, maybe there is a [bug](https://github.com/facebook/flow/issues/3717) here because it's not protecting me from referencing an invalid `status` value (which is what I intended). Well, at the very least it's good documentation and when the bug is fixed it will produce errors.